aptos-move/move-examples/marketplace/sources/coin_listing.move

```diff
@@ -0,0 +1,1080 @@
+    }
+
+    #[legacy_entry_fun]
+    public entry fun init_fixed_price_for_tokenv1<CoinType>(
+        seller: &signer,
+        token_creator: address,
+        token_collection: String,
+        token_name: String,
+        token_property_version: u64,
+        fee_schedule: Object<FeeSchedule>,
+        start_time: u64,
+        price: u64,
+    ): Object<Listing> {
+        let object = listing::create_tokenv1_container(
+            seller,
+            token_creator,
+            token_collection,
+            token_name,
+            token_property_version,
+        );
+        init_fixed_price<CoinType>(
+            seller,
+            object::convert(object),
+            fee_schedule,
+            start_time,
+            price,
+        )
+    }
+
+    #[legacy_entry_fun]
+    public entry fun init_auction<CoinType>(
+        seller: &signer,
+        object: Object<ObjectCore>,
+        fee_schedule: Object<FeeSchedule>,
+        start_time: u64,
+        starting_bid: u64,
+        bid_increment: u64,
+        auction_end_time: u64,
+        minimum_bid_time_before_end: u64,
+        buy_it_now_price: Option<u64>,
+    ): Object<Listing> {
+        let (listing_signer, constructor_ref) = init<CoinType>(
+            seller,
+            object,
+            fee_schedule,
+            start_time,
+            starting_bid,
+        );
+
+        let auction_listing = AuctionListing<CoinType> {
+            starting_bid,
+            bid_increment,
+            current_bid: option::none(),
+            auction_end_time,
+            minimum_bid_time_before_end,
+            buy_it_now_price,
+            bid_events: object::new_event_handle(&listing_signer),
+            purchase_event: object::new_event_handle(&listing_signer),
+        };
+        move_to(&listing_signer, auction_listing);
+        object::object_from_constructor_ref(&constructor_ref)
+    }
+
+    #[legacy_entry_fun]
+    public entry fun init_auction_for_tokenv1<CoinType>(
+        seller: &signer,
+        token_creator: address,
+        token_collection: String,
+        token_name: String,
+        token_property_version: u64,
+        fee_schedule: Object<FeeSchedule>,
+        start_time: u64,
+        starting_bid: u64,
+        bid_increment: u64,
+        auction_end_time: u64,
+        minimum_bid_time_before_end: u64,
+        buy_it_now_price: Option<u64>,
+    ): Object<Listing> {
+        let object = listing::create_tokenv1_container(
+            seller,
+            token_creator,
+            token_collection,
+            token_name,
+            token_property_version,
+        );
+        init_auction<CoinType>(
+            seller,
+            object::convert(object),
+            fee_schedule,
+            start_time,
+            starting_bid,
+            bid_increment,
+            auction_end_time,
+            minimum_bid_time_before_end,
+            buy_it_now_price,
+        )
+    }
+
+    inline fun init<CoinType>(
```

Comment:
Does this have to be inline?

Reply:
I see, that's fair, but we should be able to adjust if we hit compiler errors :shrug: